cmd/server.go

```diff
@@ -179,7 +184,12 @@ func loadAndVerifyServerConfig(cmd *cobra.Command) (*config.ServerConfig, error)
 		case "cert-allowed-cn":
 			conf.Security.CertAllowedCN = serverConfig.Security.CertAllowedCN
 		case "sort-dir":
-			conf.Sorter.SortDir = serverConfig.Sorter.SortDir
+			// user specified sorter dir should not take effect
+			if serverConfig.Sorter.SortDir != config.DefaultSortDir {
+				cmd.Printf(color.HiYellowString("[WARN] --sort-dir is deprecated in server settings. " +
+					"sort-dir will be set to `{data-dir}/tmp/sorter`. The sort-dir here will be no-op\n"))
+			}
+			conf.Sorter.SortDir = config.DefaultSortDir
```

Comment:
The `sorter.SortDir` should be `{data-dir}/tmp/sorter`, but it is set to `/tmp/sorter` here, will it be set later?

Reply:
yes, it will be set at `initDataDir`